src/RE.h

```diff
@@ -113,6 +114,17 @@ class Specific_RE_Matcher
 	int Match(const String* s);
 	int Match(const u_char* bv, int n);
 
+	// A disjunction is a collection of regular expressions (that under
+	// the hood are matches as a single RE, not serially) for which
+	// the match operation returns *all* of the matches.  Disjunctions
+	// are constructed using the internal "||" RE operator, and the
+	// matches are returned as indices into the position, left-to-right,
+	// of which REs matched. IMPORTANT: the first RE is numbered 1, not 0.
+	//
+	// Note that there's no guarantee regarding the ordering of the
+	// returned matches if there is more than one.
+	void MatchDisjunction(const String* s, std::vector<int>& matches);
```

Comment:
AFAICT, the new code for this is pretty much directly `MatchAll()`, but then also collecting the numbers of all accepted states. Would an overloaded `MatchAll` maybe do the job, or would you prefer to stick with the disjunction term?